test/unit/builtin_module_sources.c

```diff
@@ -0,0 +1,82 @@
+#include <stdarg.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+
+#include "trivia/util.h"
+#include "lua/builtin_modules.h"
+#include "lua/utils.h"
+#include "lua/error.h"
+
+#define UNIT_TAP_COMPATIBLE 1
+#include "unit.h"
+
+/* contents of src/lua/ files */
+extern char strict_lua[],
+	fun_lua[],
+	debug_lua[],
+	buffer_lua[],
+	errno_lua[],
+	fiber_lua[],
+	env_lua[],
+	datetime_lua[];
+
+static const char * const lua_modules[] = {
+	"strict", strict_lua,
+	"fun", fun_lua,
+	"debug", debug_lua,
+	"errno", errno_lua,
+	"fiber", fiber_lua,
+	"env", env_lua,
+	"datetime", datetime_lua,
+};
+
+static void
+builtin_sources_test(void)
+{
+	plan(56);
+	header();
+
+	struct lua_State *L = luaL_newstate();
+	luaL_openlibs(L);
+	tarantool_L = L;
+	tarantool_lua_builtin_modules_init(L);
+	tarantool_lua_error_init(L);
+	static const char * const fmts[] = {
+		"return tarantool_builtin_module('%s')",
```

Comment:
I've created simple C/Lua unit test, which is checking that result of `tarantool_builtin_module('$name')` and `tarantool_builtin_module('@builtin/$name,lua')` return expected value, as described in the `lua_modules` from `init.c`.

For that I had to refactor a bit how function is registered in LuaC thus introduced `tarantool_lua_builtin_modules_init()` function used in `libserver` code and  in the unit-test.
> I tried to use larger `tarantool_lua_init()` but it was bringing a lot of unnecessary cruft like ICU, thus used simpler/smaller approach.